media/js/base/stub-attribution.js

```diff
@@ -0,0 +1,240 @@
+     * to Bouncer download URLs as query parameters. Data returned from the service
+     * is also stored in a session cookie to save multiple requests when navigating
+     * pages. Bug https://bugzilla.mozilla.org/show_bug.cgi?id=1279291
+     */
+    var StubAttribution = {};
+
+    StubAttribution.COOKIE_CODE_ID = 'moz-stub-attribution-code';
+    StubAttribution.COOKIE_SIGNATURE_ID = 'moz-stub-attribution-sig';
+
+    /**
+     * Determines if session falls within the predefined stub attribution sample rate.
+     * @return {Boolean}.
+     */
+    StubAttribution.withinAttributionRate = function() {
+        return (Math.random() < StubAttribution.getAttributionRate()) ? true : false;
+    };
+
+    /**
+     * Returns stub attribution value used for rate limiting.
+     * @return {Number} float between 0 and 1.
+     */
+    StubAttribution.getAttributionRate = function() {
+        var rate = $('html').attr('data-stub-attribution-rate');
+        return isNaN(rate) ? 0 : Math.min(Math.max(parseFloat(rate), 0), 1);
+    };
+
+    /**
+     * Returns true if both session cookies exist.
+     * @return {Boolean} data.
+     */
+    StubAttribution.hasSessionCookie = function() {
+        return Mozilla.Cookies.hasItem(StubAttribution.COOKIE_CODE_ID) && Mozilla.Cookies.hasItem(StubAttribution.COOKIE_SIGNATURE_ID);
+    };
+
+    /**
+     * Stores a session cookie with stub attribution data values.
+     * @param {Object} data - attribution_code, attribution_sig.
+     */
+    StubAttribution.setSessionCookie = function(data) {
+
+        if (!data.attribution_code || !data.attribution_sig) {
+            return;
+        }
+
+        Mozilla.Cookies.setItem(StubAttribution.COOKIE_CODE_ID, data.attribution_code);
+        Mozilla.Cookies.setItem(StubAttribution.COOKIE_SIGNATURE_ID, data.attribution_sig);
+    };
+
+    /**
+     * Gets stub attribution data from session cookie.
+     * @return {Object} - attribution_code, attribution_sig.
+     */
+    StubAttribution.getSessionCookie = function() {
+        return {
+            /* eslint-disable camelcase */
+            attribution_code: Mozilla.Cookies.getItem(StubAttribution.COOKIE_CODE_ID),
+            attribution_sig: Mozilla.Cookies.getItem(StubAttribution.COOKIE_SIGNATURE_ID)
+            /* eslint-enable camelcase */
+        };
+    };
+
+    /**
+     * Updates all download links on the page with additional query params for
+     * stub attribution.
+     * @param {Object} data - attribution_code, attribution_sig.
+     */
+    StubAttribution.updateBouncerLinks = function(data) {
+        /**
+         * If data is missing or the browser does not meet requirements for
+         * stub attribution, then do nothing.
+         */
+        if (!data.attribution_code || !data.attribution_sig || !StubAttribution.meetsRequirements()) {
+            return;
+        }
+
+        $('a[href*="https://download.mozilla.org/"]').each(function() {
```

Comment:
I've added this now. I made the call to append the params to both 32 and 64bit Windows URLs. The 64bit links don't seem to use the stub installer currently, but I'm making the assumption this could change in the future.